kube-runtime/src/controller/runner.rs

```diff
@@ -102,12 +109,35 @@ where
             Poll::Pending => {}
         }
         loop {
+            // If we are at our limit or not ready to start executing, then there's
+            // no point in trying to get something from the scheduler, so just put
+            // all expired messages emitted from the queue into pending.
+            if (*this.max_concurrent_executions > 0
+                && slots.len() >= *this.max_concurrent_executions as usize)
+                || !*this.is_ready_to_execute
+            {
+                match scheduler.as_mut().hold().poll_next_unpin(cx) {
+                    Poll::Pending => break Poll::Pending,
+                    Poll::Ready(None) => {
+                        break if has_active_slots {
```

Comment:
it isn't tested since Hold only returns `Poll::Ready(None)` when the requests stream is over, i.e. `scheduler.requests.poll_next()` returns `Poll::Ready(None)`. i'll modify the test to drop the sender in order for this block to get tested as well 👍